zingo-status/src/confirmation_status.rs

```diff
@@ -0,0 +1,130 @@
+use zcash_primitives::consensus::BlockHeight;
+/// Transaction confirmation states. Every transaction record includes exactly one of these variants.
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ConfirmationStatus {
+    /// The transaction has been broadcast to the zcash blockchain. It may be waiting in the mempool.
+    /// The height of the chain as the transaction was broadcast.
+    /// # Examples
+    ///
+    /// ```
+    /// use zingo_status::confirmation_status::ConfirmationStatus;
+    /// use zcash_primitives::consensus::BlockHeight;
+    ///
+    /// let status = ConfirmationStatus::Broadcast(BlockHeight::from_u32(100));
+    /// assert_eq!(status.is_broadcast(), true);
+    /// assert_eq!(status.get_height(), BlockHeight::from_u32(100));
+    /// ```
+    Broadcast(BlockHeight),
+    /// The transaction has been included in at-least one block mined to the zcash blockchain.
+    /// The height of a confirmed block that contains the transaction.
+    /// # Examples
+    ///
+    /// ```
+    /// use zingo_status::confirmation_status::ConfirmationStatus;
+    /// use zcash_primitives::consensus::BlockHeight;
+    ///
+    /// let status = ConfirmationStatus::Confirmed(BlockHeight::from_u32(200));
+    /// assert_eq!(status.is_confirmed(), true);
+    /// assert_eq!(status.get_height(), BlockHeight::from_u32(200));
+    /// ```
+    Confirmed(BlockHeight),
+}
+
+impl ConfirmationStatus {
+    /// Converts a blockheight and unconfirmed status into a `ConfirmationStatus`.
+    /// # Examples
+    ///
+    /// ```
+    /// use zingo_status::confirmation_status::{ConfirmationStatus, BlockHeight};
+    ///
+    /// let status = ConfirmationStatus::from_blockheight_and_unconfirmed_bool(BlockHeight(10), false);
+    /// assert_eq!(status, ConfirmationStatus::Confirmed(BlockHeight(10)));
+    ///
+    /// let status = ConfirmationStatus::from_blockheight_and_unconfirmed_bool(BlockHeight(10), true);
+    /// assert_eq!(status, ConfirmationStatus::Broadcast(BlockHeight(10)));
+    /// ```
+    pub fn from_blockheight_and_unconfirmed_bool(
+        blockheight: BlockHeight,
+        unconfirmed: bool,
+    ) -> Self {
+        if unconfirmed {
+            Self::Broadcast(blockheight)
+        } else {
+            Self::Confirmed(blockheight)
+        }
+    }
+    pub fn is_broadcast(&self) -> bool {
+        matches!(self, Self::Broadcast(_))
+    }
+    pub fn is_confirmed(&self) -> bool {
+        matches!(self, Self::Confirmed(_))
+    }
```

Comment:
There're overlapping cases here.   Two different programmers could use this API can call two different calls that return true when the BlockHeight is "at"...   then later in life that calling code could have divergent intention.   Are these methods already called somewhere?

If they're not already explicitly used then I vote we remove them.

If they are used, then please unify the API so there's only one way to do a thing.

Reply:
calling code can have different intentions: intended